### PR TITLE
refactor(ui): Generalize FilterChips for reuse in other components

### PR DESCRIPTION
**Pull request by dvail**
## Description

Refactors the `FilterChips` component so that it is more general for reuse in other sections of the app.
image

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

## Testing Performed

Un-skip the `workloadTableToolbar.test.js` test file and run locally...
image

Manual testing of Workload CVE filters:
image
image
image
image

Manual testing of override for `isGlobal` prop.
image

### CODE REVIEW

**Comment by dvail** [comment]
Current dependencies on/for this PR:
* master
  * **PR #7021** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/7021" target="_blank">Graphite</a> 
    * **PR #7024** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/7024" target="_blank">Graphite</a> 
      * **PR #7054** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/7054" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/stackrox/stackrox/7054?utm_source=stack-comment).

File: ui/apps/platform/src/Components/PatternFly/SearchFilterChips.tsx [hunk 1]
```diff
@@ -0,0 +1,100 @@
+import React from 'react';
+import { Flex, FlexItem, ChipGroup, Chip, Button, ToolbarChip } from '@patternfly/react-core';
```
Comment (pedrottimark) [inline ui/apps/platform/src/Components/PatternFly/SearchFilterChips.tsx:2]: Small request to order named imports. Sometimes auto additions by editor are too invisible.

Lesser importance, but you might also order by hooks, types, utils below.

File: ui/apps/platform/src/Components/PatternFly/SearchFilterChips.tsx [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    filterChipGroupDescriptors: FilterChipGroupDescriptor[];
+};
+
+/**
+ * Displays and manages the search filter chips for a given set of search filter categories based
+ * on the current URL search filter
+ */
+function SearchFilterChips({
+    onFilterChange = noop,
+    filterChipGroupDescriptors,
```
Comment (pedrottimark) [inline ui/apps/platform/src/Components/PatternFly/SearchFilterChips.tsx:33]: Maybe also list in alpha order, or maybe more important, obligatory property before optional?

File: ui/apps/platform/src/Components/PatternFly/SearchFilterChips.tsx [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    function onChangeSearchFilter(newFilter: SearchFilter) {
+        setSearchFilter(newFilter);
+        onFilterChange(newFilter);
+    }
+
+    function onDelete(category: string, chip: ToolbarChip | string) {
+        const newSearchFilter = { ...searchFilter };
+        const newResourceFilter = searchFilter[category] as string[];
+        const chipKey = typeof chip === 'string' ? chip : chip.key;
+        newSearchFilter[category] = newResourceFilter.filter((fil: string) => fil !== chipKey);
```
Comment (pedrottimark) [inline ui/apps/platform/src/Components/PatternFly/SearchFilterChips.tsx:46]: Ha ha, so many meanings of filter in one line ;)

File: ui/apps/platform/src/Components/PatternFly/SearchFilterChips.tsx [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+    const { searchFilter, setSearchFilter } = useURLSearch();
+
+    function onChangeSearchFilter(newFilter: SearchFilter) {
+        setSearchFilter(newFilter);
+        onFilterChange(newFilter);
+    }
+
+    function onDelete(category: string, chip: ToolbarChip | string) {
+        const newSearchFilter = { ...searchFilter };
+        const newResourceFilter = searchFilter[category] as string[];
```
Comment (pedrottimark) [inline ui/apps/platform/src/Components/PatternFly/SearchFilterChips.tsx:44]: Maybe my mind is slowing down too fast on a Friday. What is intended meaning of Resource?

Besides that it is a term we are trying (painfully slowly) to supersede, this seems like a re-usable component that is not limited to resources, even if that is the original use.

**Review by pedrottimark — APPROVED** [review]
Thank you for refactoring. A few comments, possibly confused to some extent :)

If possible, some other team members who might use sooner than me can also review.

File: ui/apps/platform/src/Components/PatternFly/SearchFilterChips.tsx (hunk 4 above)
Comment (dvail) [reply ui/apps/platform/src/Components/PatternFly/SearchFilterChips.tsx:44]: Good point - resource here doesn't make sense in generic component.

File: ui/apps/platform/src/Components/PatternFly/SearchFilterChips.tsx [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+ */
+function SearchFilterChips({
+    onFilterChange = noop,
+    filterChipGroupDescriptors,
+}: SearchFilterChipsProps) {
+    const { searchFilter, setSearchFilter } = useURLSearch();
+
+    function onChangeSearchFilter(newFilter: SearchFilter) {
+        setSearchFilter(newFilter);
+        onFilterChange(newFilter);
```
[the remark this answers is not in the document]
Comment (dvail) [reply ui/apps/platform/src/Components/PatternFly/SearchFilterChips.tsx:39]: The main idea here was to expose the ability for a parent component to get the updated filter when it changes without needing to parse the URL. Although not in use at this point, it felt low-effort to include it as a similar pattern to other event callbacks.

**Comment by dvail** [comment]
@alwayshooin @sachaudh since you both either have recently worked on, or are currently working on VM 2.0 can you take a quick look at this too?

Also Linda I know default filters are disabled at the moment, but I did some manual testing and this will support the `isGlobal` prop used for custom rendering.

**Pull request merged**